src/main/java/org/wso2/carbon/apimgt/securityenforcer/publisher/HttpDataPublisher.java

```diff
@@ -65,13 +68,27 @@ public HttpDataPublisher(AISecurityHandlerConfig.AseConfig aseConfiguration,
         httpClient = SecurityUtils.getHttpClient(protocol, dataPublisherConfiguration);
         setAuthToken(aseConfiguration.getAseToken());
         setEndPoint(aseConfiguration.getEndPoint());
+        aseConfig = aseConfiguration;
+        dataPublisherConfig = dataPublisherConfiguration;
     }
 
     public HttpDataPublisher(String endPoint, String authToken) {
         setAuthToken(authToken);
         setEndPoint(endPoint);
     }
 
+    private void reCreateDataPublisherInstance() {
+        String protocol;
+        try {
+            protocol = new URL(aseConfig.getEndPoint()).getProtocol();
+            httpClient = SecurityUtils.getHttpClient(protocol, dataPublisherConfig);
```

Comment:
We cannot use the same httpClient, because old httpClient was created with a different ASE end point.
When data publish to ASE fails, the ASE data end point will be shifted and a new httpClient must be created.

Reply:
The client is independent from the endpoint. The only thing it needs is the endpoint protocol. Since we are recommending to use https, that won't be an issue.

Reply:
Changed the code to not create DataPublisher instance. Instead just resetting endpoint.